### PR TITLE
23 support specification of custom vertical flow profiles in ini file

### PR DESCRIPTION
**Pull request by erwin-mulder**
under the [sealock] header, the user can specify the flow_profile key with one or more comma separated values. These values do not have to be normalized, this will be done internally by libzsf.

In this change I added support for reading a line of comma separated double values from the zsf_config ini file.
I also refactored and improved code, so usability and consistency was been improved.

Unit testing has been added for all the ini_read functions.

### CODE REVIEW

File: src/zsf_config.c [hunk 1]
```diff
@@ -41,36 +39,54 @@ int _zsf_ini_handler(char *section, char *key, char *value, void *data_ptr) {
...
     } else if (!strcmp(key, "lock_width")) {
-      config_ptr->locks[lock_index].parameters.lock_width = strtod(value, &end_ptr);
+      config_ptr->locks[lock_index].parameters.lock_width = ini_parse_double(value, &status);
     } else if (!strcmp(key, "lock_bottom")) {
-      config_ptr->locks[lock_index].parameters.lock_bottom = strtod(value, &end_ptr);
+      config_ptr->locks[lock_index].parameters.lock_bottom = ini_parse_double(value, &status);
+    } else if (!strcmp(key, "flow_profile")) {
+      int array_length = 0;
+      double *value_array = ini_parse_double_list(value, &array_length, &status);
+      if (array_length && value_array && status == INI_OK) {
```
Comment (vrielin) [inline src/zsf_config.c]: Would it also be possible to check only for != INI_FAIL?

File: tests/test_ini_read.c [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  int status = ini_read(filepath, my_ini_handler, NULL);
+  TEST_ASSERT_MESSAGE(status == INI_OK, "Failed to load INI file.");
+}
+
+static void test_ini_parse_int() {
+  int status = INI_OK;
+  int ival = 0;
+
+  ival = ini_parse_int("42", &status);
+  TEST_ASSERT_MESSAGE(status == INI_OK, "Conversion to int failed.");
```
Comment (vrielin) [inline tests/test_ini_read.c:72]: There is no INI_FAIL test (but I am fine with that).

File: tests/test_ini_read.c [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  dval = ini_parse_double("-567.8", &status);
+  TEST_ASSERT_MESSAGE(status == INI_OK, "Conversion to double failed.");
+  TEST_ASSERT_DOUBLE_WITHIN(0.05, -567.8, dval);
+}
+
+static void test_ini_parse_double_list() {
+  int status = INI_OK;
+  int length = 0;
+  double *array = ini_parse_double_list("1,1,2,3.8, 5 ,8", &length, &status);
```
Comment (vrielin) [inline tests/test_ini_read.c:125]: What happens if the list is empty? Is that an easy way to add a test for status == INI_FAIL?

**Review by vrielin — APPROVED** [review]

File: src/zsf_config.c (hunk 1 above)
Comment (erwin-mulder) [reply src/zsf_config.c]: Agreed:
First two checks can be removed.
Also: in unit test we need to check handling of empty list.

File: tests/test_ini_read.c (hunk 2 above)
Comment (erwin-mulder) [reply tests/test_ini_read.c:72]: Will add 'failing' conversion test.

File: tests/test_ini_read.c (hunk 3 above)
Comment (erwin-mulder) [reply tests/test_ini_read.c:125]: Will add 'failing' conversion test and (also) explicitly an 'empty' list.
